modelql-core/src/commonMain/kotlin/org/modelix/modelql/core/IStep.kt

```diff
@@ -34,7 +34,7 @@ interface IStep {
 
     fun requiresWriteAccess(): Boolean = false
     fun hasSideEffect(): Boolean = requiresWriteAccess()
-    fun needsCoroutineScope() = false
+    fun needsCoroutineScope() = hasSideEffect()
```

Comment:
It would probably be good to explain somewhere, why needing a couroutine scope depends on the side effect.